karton/yaramatcher/yaramatcher.py

```diff
@@ -110,62 +107,20 @@ def scan_sample(self, sample: bytes) -> List[str]:
         return rule_names
 
-    def process_cuckoo(self, task: Task) -> List[str]:
+    def process_analysis(self, task: Task) -> List[str]:
+        self.log.info("Processing analysis")
         yara_matches: List[str] = []
-        analysis = task.get_payload("analysis")
-        self.log.info(f"Processing cuckoo analysis {analysis.name}")
-
-        with analysis.extract_temporary() as analysis_dir:
-            dump_dir = f"{analysis_dir}/dumps"
-            for rootdir, _dirs, files in os.walk(dump_dir):
-                for filename in files:
-                    if filename.endswith(".txt") or filename.endswith(".metadata"):
-                        continue
-                    self.log.debug(f"Checking {filename}")
-                    with open(f"{rootdir}/{filename}", "rb") as dumpf:
-                        content = dumpf.read()
-                    yara_matches += self.scan_sample(content)
-
-        return yara_matches
-
-    def process_drakrun(self, task: Task) -> List[str]:
-        self.log.info("Processing drakrun analysis")
-        yara_matches: List[str] = []
-
-        with tempfile.TemporaryDirectory() as tmpdir:
-            dumpsf = os.path.join(tmpdir, "dumps.zip")
-            task.get_resource("dumps.zip").download_to_file(dumpsf)  # type: ignore
-
-            zipf = zipfile.ZipFile(dumpsf)
-            zipf.extractall(tmpdir)
-
-            for rootdir, _dirs, files in os.walk(tmpdir):
-                for filename in files:
-                    # skip non-dump files
-                    if not re.match(r"^[a-f0-9]{4,16}_[a-f0-9]{16}$", filename):
-                        continue
 
-                    with open(f"{rootdir}/{filename}", "rb") as dumpf:
-                        content = dumpf.read()
-                    yara_matches += self.scan_sample(content)
-
-        return yara_matches
-
-    def process_joesandbox(self, task: Task) -> List[str]:
-        self.log.info("Processing joesandbox analysis")
-        yara_matches: List[str] = []
-
-        with tempfile.TemporaryDirectory() as tmpdir:
-            dumpsf = os.path.join(tmpdir, "dumps.zip")
-            task.get_resource("dumps.zip").download_to_file(dumpsf)  # type: ignore
+        dumps = task.get_resource("dumps.zip")
+        dumps_metadata = task.get_payload("dumps_metadata")
 
-            zipf = zipfile.ZipFile(dumpsf)
-            zipf.extractall(tmpdir, pwd=b"infected")
-
-            for rootdir, _dirs, files in os.walk(tmpdir):
-                for filename in files:
-                    with open(f"{rootdir}/{filename}", "rb") as dumpf:
-                        content = dumpf.read()
+        with dumps.extract_temporary() as tmpdir:  # type: ignore
+            for dump_metadata in dumps_metadata:
+                dump_path = (Path(tmpdir) / Path(dump_metadata["filename"])).resolve()
+                if not str(dump_path).startswith(tmpdir):
+                    continue
+                with open(dump_path, "rb") as dumpf:
```

Comment:
If we're using pathlib and passing the content we might as well use `path.get_bytes()` here IMO

Reply:
Done